tests/clients/test_pep0249.py

```diff
@@ -0,0 +1,320 @@
+import logging
+from typing import Generator, TYPE_CHECKING
+from unittest.mock import patch
+import psycopg2
+import psycopg2.extras
+import pytest
+from instana.instrumentation.pep0249 import (
+    ConnectionFactory,
+    ConnectionWrapper,
+    CursorWrapper,
+)
+from opentelemetry.trace import SpanKind
+from instana.singletons import tracer
+from instana.util.traceutils import get_tracer_tuple
+from pytest import LogCaptureFixture
+from instana.span.span import InstanaSpan
+
+
+class TestCursorWrapper:
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
+        self.connect_params = [
+            "db",
+            {
+                "db": "instana_test_db",
+                "host": "localhost",
+                "port": "5432",
+                "user": "root",
+                "password": "passw0rd",
+            },
+        ]
+        self.test_conn = psycopg2.connect(
+            database=self.connect_params[1]["db"],
+            host=self.connect_params[1]["host"],
+            port=self.connect_params[1]["port"],
+            user=self.connect_params[1]["user"],
+            password=self.connect_params[1]["password"],
+        )
+        self.cursor_params = {"key": "value"}
+        self.test_cursor = self.test_conn.cursor()
+        self.cursor_name = "test-cursor"
+        self.test_wrapper = CursorWrapper(
+            self.test_cursor,
+            self.cursor_name,
+            self.connect_params,
+            self.cursor_params,
+        )
+        self.test_cursor.execute(
+            """
+            DROP TABLE IF EXISTS tests;
+            CREATE TABLE tests (id SERIAL PRIMARY KEY, name VARCHAR(50), email VARCHAR(100));
+            """
+        )
+        self.test_conn.commit()
+        self.test_cursor.execute(
+            """
+            INSERT INTO tests (id, name, email) VALUES (1, 'test-name', 'testemail@mail.com');
+            """
+        )
+        self.test_conn.commit()
+        self.test_cursor.execute("""
+            DROP PROCEDURE IF EXISTS insert_user(IN test_id INT, IN test_name VARCHAR, IN test_email VARCHAR);
+            CREATE PROCEDURE insert_user(IN test_id INT, IN test_name VARCHAR, IN test_email VARCHAR)
+            LANGUAGE plpgsql
+            AS $$
+            BEGIN
+                INSERT INTO tests (id, name, email) VALUES (test_id, test_name, test_email);
+            END;
+            $$;
+            """)
```

Comment:
I am not comfortable with all this setting up here. 
You are running the `cursor()`, `execute()` and `commit()` during the setup of each test case. Think on having just the necessary background objects to be able to execute your tests.

Reply:
Objects has been divided into different functions as reset_table and reset procedure